core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -272,6 +277,74 @@ public int getCapacity()
         return hashCapacity;
     }
 
+    private int putIfAbsent(int firstPosition, int positions, Page page, @Nullable BlockBuilder groupIdsBuilder)
```

Comment:
Would that make sense to implement `private int putIfAbsent(int position, Page page, long rawHash)` as a call to new `putIfAbsent`? Or creating local arrays would make this approach performance too poor?

Reply:
> Or creating local arrays would make this approach performance too poor?

I expect it to reduce performance, but there are no dictionary aggregation benchmark in `BenchmarkHashAndStreamingAggregationOperators` so it's hard to say.